File: algorithms/1_easy/C/02520_count_the_digits_that_divide_a_number/solution.py
class Solution:
    def countDigits(self, num: int) -> int:
        num_cp = num
        count = 0

        while num_cp > 0:
            dig = num_cp % 10
            num_cp //= 10

            if dig == 0:
                continue

            if num % dig == 0:
                count += 1

        return count

File: algorithms/1_easy/C/02520_count_the_digits_that_divide_a_number/solution.rs
impl Solution {
    pub fn count_digits(num: i32) -> i32 {
        let mut num_cp = num;
        let mut count = 0_i32;

        while num_cp > 0 {
            let dig = num_cp % 10;
            num_cp /= 10;

            if dig == 0 {
                continue;
            }

            if num % dig == 0 {
                count += 1;
            }
        }

        count
    }
}
